drivers/adc/adc_shell.c

```diff
@@ -233,256 +214,198 @@ static int cmd_adc_acq(const struct shell *shell, size_t argc, char **argv)
+
+	adc->channel_config.channel_id = adc_channel_id;
 	const struct adc_sequence sequence = {
-		.channels	=
-			BIT(adc_list[chosen_adc].channel_config.channel_id),
+		.channels	= BIT(adc->channel_config.channel_id),
 		.buffer		= m_sample_buffer,
 		.buffer_size	= sizeof(m_sample_buffer),
-		.resolution	= adc_list[chosen_adc].resolution,
+		.resolution	= adc->resolution,
 	};
+
 	retval = adc_read(adc_dev, &sequence);
 	if (retval >= 0) {
-		shell_fprintf(shell, SHELL_NORMAL,
-				"Read: %i\n", m_sample_buffer[0]);
+		shell_print(shell, "Read: %i", m_sample_buffer[0]);
 	}
+
 	return retval;
 }
 
 static int cmd_adc_print(const struct shell *shell, size_t argc, char **argv)
 {
-	int chosen_adc = -1;
-	int i;
-	char *gain = "1";
-	char *ref = "INTERNAL";
-	uint16_t acq_time;
-	uint8_t channel_id;
-	uint8_t resolution;
+	/* -1 index of ADC label name */
+	struct adc_hdl *adc = get_adc(argv[-1]);
+	char *gain = "NA";
+	char *ref = "NA";
 
-	chosen_adc = get_adc_from_list(argv[args_indx.adc]);
-	if (chosen_adc < 0) {
-		shell_error(shell, "Device not in device list");
-		return 0;
-	}
-	for (i = 0; i < ARRAY_SIZE(gain_list); i++) {
-		if (gain_list[i].gain ==
-				adc_list[chosen_adc].channel_config.gain) {
+	for (int i = 0; i < ARRAY_SIZE(gain_list); i++) {
+		if (gain_list[i].gain == adc->channel_config.gain) {
 			gain = gain_list[i].string;
 		}
 	}
-	for (i = 0; i < ARRAY_SIZE(reference_list); i++) {
+
+	for (int i = 0; i < ARRAY_SIZE(reference_list); i++) {
 		if (reference_list[i].reference ==
-				adc_list[chosen_adc].channel_config.reference) {
+				adc->channel_config.reference) {
 			ref = reference_list[i].string;
 		}
 	}
-	acq_time = adc_list[chosen_adc].channel_config.acquisition_time;
-	channel_id = adc_list[chosen_adc].channel_config.channel_id;
-	resolution = adc_list[chosen_adc].resolution;
-	shell_fprintf(shell, SHELL_NORMAL, "%s:\n"
-			"Gain: %s\n"
-			"Reference: %s\n"
-			"Acquisition Time: %u\n"
-			"Channel ID: %u\n"
-			"Resolution: %u\n",
-			argv[args_indx.adc],
-			gain,
-			ref,
-			acq_time,
-			channel_id,
-			resolution);
+
+	shell_print(shell, "%s:\n"
+			   "Gain: %s\n"
+			   "Reference: %s\n"
+			   "Acquisition Time: %u\n"
+			   "Channel ID: %u\n"
+			   "Resolution: %u",
+			   adc->device_label,
+			   gain,
+			   ref,
+			   adc->channel_config.acquisition_time,
+			   adc->channel_config.channel_id,
+			   adc->resolution);
 	return 0;
 }
+
 SHELL_STATIC_SUBCMD_SET_CREATE(sub_ref_cmds,
-	/* Alphabetically sorted. */
-	SHELL_CMD(VDD_1, NULL, "VDD", cmd_adc_ref),
-	SHELL_CMD(VDD_1_2, NULL, "VDD/2", cmd_adc_ref),
-	SHELL_CMD(VDD_1_3, NULL, "VDD/3", cmd_adc_ref),
-	SHELL_CMD(VDD_1_4, NULL, "VDD/4", cmd_adc_ref),
-	SHELL_CMD(INT, NULL, "Internal", cmd_adc_ref),
-	SHELL_CMD(EXT0, NULL, "EXT0", cmd_adc_ref),
-	SHELL_CMD(EXT1, NULL, "EXT1", cmd_adc_ref),
-	SHELL_SUBCMD_SET_END /* Array terminated. */
+	SHELL_CMD_ARG(VDD_1, NULL, "VDD", cmd_adc_ref, 1, 0),
```

Comment:
we could think at some point about shell extension which would allow creating subcommand with a dictionary. In that case argv would contain direct value. Something like
```
SHELL_STATIC_SUBCMD_DICT_CREATE(sub, dictionary_array)
```
where array contains string + value. String for tab handle, value passed to command handler. That would make code much compact in cases like this and there are many cases like this one.

Reply:
May you please create an issue for that?